server/controllers/finance/reports/reportAccounts/index.js

```diff
@@ -63,47 +59,41 @@ function document(req, res, next) {
  * This feature select all transactions for a specific account
 */
 function queryReportAccount(accountId, source) {
-  source = parseInt(source);
-  let sql;
-
-  if (source === 1) {
-    sql = `
-      SELECT general_ledger.trans_id, BUID(general_ledger.entity_uuid) AS entity_uuid, general_ledger.description,
-      general_ledger.trans_date, general_ledger.debit_equiv as debit, general_ledger.credit_equiv as credit
-      FROM general_ledger
-      WHERE general_ledger.account_id = ?
-      GROUP BY general_ledger.trans_id
-      ORDER BY general_ledger.trans_date ASC;
-    `;
-  } else if (source === 2) {
-    sql = `
-      SELECT posting_journal.trans_id, BUID(posting_journal.entity_uuid) AS entity_uuid, posting_journal.description,
-      posting_journal.trans_date, posting_journal.debit_equiv as debit, posting_journal.credit_equiv as credit
-      FROM posting_journal
-      WHERE posting_journal.account_id = ?
-      GROUP BY posting_journal.trans_id
-      ORDER BY posting_journal.trans_date ASC;
+  const sourceId = parseInt(source, 10);
+
+  // get the table name
+  const tableName = sourceMap[sourceId];
+
+  const sql = `
+      SELECT trans_id, BUID(entity_uuid) AS entity_uuid, description, trans_date, 
+        debit_equiv as debit, credit_equiv as credit
+      FROM ${tableName}
+      WHERE account_id = ?
+      GROUP BY trans_id 
+      ORDER BY trans_date ASC`;
+
+  const sqlAggrega = ` SELECT SUM(t.debit) AS debit, SUM(t.credit) AS credit, SUM(t.debit - t.credit) AS balance 
+    FROM (
+      SELECT trans_id, BUID(entity_uuid) AS entity_uuid, description, trans_date, 
+        debit_equiv as debit, credit_equiv as credit
+      FROM ${tableName}
+      WHERE account_id = ?
+      GROUP BY trans_id 
+      ORDER BY trans_date ASC
+    ) AS t 
     `;
-  } else if (source === 3) {
-    sql = `
-      SELECT transaction.trans_id, transaction.entity_uuid, transaction.description, transaction.trans_date, sum(transaction.credit_equiv) as credit, sum(transaction.debit_equiv) as debit
-      FROM(
-        SELECT posting_journal.trans_id, BUID(posting_journal.entity_uuid) AS entity_uuid, posting_journal.description,
-        posting_journal.trans_date, posting_journal.debit_equiv, posting_journal.credit_equiv
-        FROM posting_journal
-        WHERE posting_journal.account_id = ?
-        UNION
-        SELECT general_ledger.trans_id, BUID(general_ledger.entity_uuid) AS entity_uuid, general_ledger.description,
-        general_ledger.trans_date, general_ledger.debit_equiv, general_ledger.credit_equiv
-        FROM general_ledger
-        WHERE general_ledger.account_id = ?
-      ) as transaction
-      GROUP BY transaction.trans_id
-      ORDER BY transaction.trans_date ASC;
-    `;
-  }
 
-  return db.exec(sql, [accountId, accountId]);
+  const bundle = {};
+
+  return db.exec(sql, [accountId, accountId])
```

Comment:
This code is correct. :+1:

Since the promises do not depend on one another, you can actually do both at the same time to save space and make the code (slightly) faster.

Like this:
```js

const params = [ accountId, accountId ];
return q.all([ db.exec(sql, params), db.exec(sql, params) ]);

// in the main function
 promise.spread((accounts, sum)) => {
  _.extend(bundle, { accounts, sum });
  // ... etc ...
 });
```